### ISSUE TITLE
Sharing map with features from certain layer types selected not working

### ISSUE THREAD

**Issue by tonnyandersson** [origo-map/origo#969]
**Describe the bug**
Sharing a map with group layer features selected triggers an error and the map crashes when loading the shared map.

**To Reproduce**
1. Create a map with a group layer (layer type "GROUP")
2. Load the map and select a feature in the group layer
3. Share the map
4. Open the shared map in a new tab/window

**Expected behavior**
The map should load without issues, zooming to the selected feature.

**Comment by tonnyandersson** [comment]
Sharing a map with multiple layers with the same layer name also produces a similar result. I.e. multiple layers from the same source layer, specified with double underscore suffixes in the config file. Example:

```
{
  "name": "mylayer__1",
  "title": "My Layer 1",
  "source": "myDataSource"
  ...
},
{
  "name": "mylayer__2",
  "title": "My Layer 2",
  "source": "myDataSource"
  ...
}
```

**Comment by Grammostola** [comment]
I've had a look and think I've a prelim solution for group layers (I haven't tested all kinds of trees but a few basic ones with grouplayers in grouplayers seem ok). 

 "__"-wfs-layers are a different species of fish, as you mentioned. (I wasn't able to repro that issue with a local geojson layer).  Possibly there's a narrow path that needs to be found when handling layer names and feature ids and viewer and probably permalink seem involved. Keep them fishes as one issue?

I'll continue digging anyways.

**Comment by tonnyandersson** [comment]
Yeah, I believe it's all connected. How layer names are handled, how they are split and transformed into id:s, settin' and gettin' permalinks and selecting items for feature info stuff... It's all connected.

I guess layer groups are not much of an issue if you stick to unique group and layer names. In such cases, all you have to do is to check for group layers and then dive into them to find the layers within, or something like that. However, problems will arise if you use the same naming scheme as in the [release test config](https://www.malardalskartan.se/origo/v2.2.0/test.json), e.g.:

```
{
  "name": "layername",
  "type": "GROUP",
  "layers": [
    {
      "name": "layername__1",
      "type": "WFS",
      ...
    },
    {
      "name": "layername__2",
      "type": "WFS",
      ...
    }
  ]
}
```

I know, this is essentially an _underscoreunderscore_ issue rather than a group layer issue. But I think this is a not so uncommon scenario, grouping several filtered layers from the same source layer in the same group.

Perhaps we should treat them as separate issues, just to get this group layer issue out of the way. But that feels like half-assing it.

**Comment by Grammostola** [comment]
I see the point, that they are pretty related. Implementation-wise layers-in-groups seems cleaner and in one place, as opposed to the __ thing, which I've made some limited progress on only by controversial changes in several places(it seems to be a layer name, layer id, feature id thing). Perhaps two PRs targeting this issue is the way. Will try to produce those this week but the second is more uncertain and both "__"-scenarios in the docs need to work.

**Comment by jokd** [comment]
One part of this issue is solved with #1200 but the __-layers haven't been handled yet.
